bld/CLMBuildNamelist.pm

```diff
@@ -768,6 +777,31 @@ sub setup_cmdl_fates_mode {
   }
 }
 
+#-------------------------------------------------------------------------------
+sub setup_cmdl_configuration_and_structure {
+   # Error-check and set the 'configuration' and 'structure' namelist flags
+
+   my ($opts, $nl_flags, $definition, $defaults, $nl) = @_;
+
+   my $val;
+
+   my $var = "configuration";
+   $val = $opts->{$var};
+   if ($val eq "clm" || $val eq "nwp") {
+      $nl_flags->{$var} = $val;
```

Comment:
I was confused about the difference between opts and nl_flags. It looks like most/all opts are also added to nl_flags, then accessed from there; is that right? i.e., is what I do in setup_cmdl_configuration_and_structure reasonable?

Reply:
$opts is treated read-only as it's the command line arguments. nl_flags are any flags or values worth saving as the processing goes on. So this looks correct to me.